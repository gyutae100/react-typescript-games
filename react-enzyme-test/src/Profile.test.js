import React from 'react';
import { mount } from 'enzyme';
import Profile from './Profile';

describe('<Profile />', ()=>{
    it('matches snapshot', ()=>{
        const wrapper = mount(<Profile username="kim" name="gyutae" />);
        expect(wrapper).toMatchSnapshot();
    })

    it('renders username and name',()=>{
        const wrapper = mount(<Profile username="kim" name="gyutae" />);
        
        //props확인
        expect(wrapper.props().username).toBe('kim');
        expect(wrapper.props().name).toBe('gyutae')
    
        //dom 확인
        const boldElement =wrapper.find('b');
        expect(boldElement.contains('kim')).toBe(true);
        const spanElement = wrapper.find('span');
        expect(spanElement.text()).toBe('gyutae');

    })
})