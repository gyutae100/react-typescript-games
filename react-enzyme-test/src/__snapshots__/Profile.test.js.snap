// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<Profile
  name="gyutae"
  username="kim"
>
  <div>
    <b>
      kim
    </b>
    <span>
      gyutae
    </span>
  </div>
</Profile>
`;
